pkg/provider/azure_local_services.go

```diff
@@ -544,3 +543,84 @@ func (az *Cloud) cleanupLocalServiceBackendPool(
 	}
 	return lbs, nil
 }
+
+// checkAndApplyLocalServiceBackendPoolUpdates if the IPs in the backend pool are aligned
+// with the corresponding endpointslice, and update the backend pool if necessary.
+func (az *Cloud) checkAndApplyLocalServiceBackendPoolUpdates(lb network.LoadBalancer, service *v1.Service) error {
+	serviceName := getServiceName(service)
+	endpointsNodeNames, err := az.getLocalServiceEndpointsNodeNames(service)
+	if err != nil {
+		return err
+	}
+	var expectedIPs []string
```

Comment:
how about using Set here? in that way, you could compare by Set.Equal before L573 and skip reconciling if the Sets are not changed.

Reply:
In this case it will be skipped in L616.